### PR TITLE
Use new 1.22.3 builder

### PR DESCRIPTION
**Pull request by akalenyu**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
Golang 1.22.3 builder is now used
```

### CODE REVIEW

File: hack/ci/prom_metric_linter.sh [hunk 1]
```diff
@@ -49,7 +43,7 @@ while [[ $# -gt 0 ]]; do
 done
 
 # Get the metrics list
-go build -o _out/prom-metrics-collector "$METRICS_COLLECTOR_PATH/..."
+./hack/build/bazel-docker.sh go build -o _out/prom-metrics-collector /root/go/src/kubevirt.io/containerized-data-importer/tools/prom-metrics-collector
```
Comment (akalenyu) [inline hack/ci/prom_metric_linter.sh:46]: @avlitman Hey, would appreciate an ack on this commit - https://github.com/kubevirt/containerized-data-importer/pull/3250/commits/db308632d4bb92e3730a1ee028275cc5902d4fee

Comment (avlitman) [reply hack/ci/prom_metric_linter.sh:46]: looks nice thanks! 
Just to be sure did you change a metric name to not start with `kubevirt_cdi` for example, then run `make lint-metrics` and see it fail/pass as expected?

**Comment by akalenyu** [comment]
> @akalenyu: The following test **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:
> 
> Test name	Commit	Details	Required	Rerun command
> pull-containerized-data-importer-e2e-ceph	[cd4ceb6](https://github.com/kubevirt/containerized-data-importer/commit/cd4ceb619bd54cd7f6e659b669a6463d0d15736b)	[link](https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_containerized-data-importer/3250/pull-containerized-data-importer-e2e-ceph/1790361731785035776)	true	`/test pull-containerized-data-importer-e2e-ceph`

So there's definitely a bug somewhere since the image in this test should carry those environment variables.
```
Expected
    <map[string]string | len:5>: {
        "alerts.k8s.io/KubePersistentVolumeFillingUp": "disabled",
        "app": "containerized-data-importer",
        "app.kubernetes.io/component": "storage",
        "app.kubernetes.io/managed-by": "cdi-controller",
        "test.kubevirt.io/existing": "existing",
    }
to have {key: value}
    <map[interface {}]interface {} | len:1>: {
        <string>"test.kubevirt.io/test": <string>"testvalue",
```

**Comment by akalenyu** [comment]
> > @akalenyu: The following test **failed**, say `/retest` to rerun all failed tests or `/retest-required` to rerun all mandatory failed tests:
> > Test name	Commit	Details	Required	Rerun command
> > pull-containerized-data-importer-e2e-ceph	[cd4ceb6](https://github.com/kubevirt/containerized-data-importer/commit/cd4ceb619bd54cd7f6e659b669a6463d0d15736b)	[link](https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_containerized-data-importer/3250/pull-containerized-data-importer-e2e-ceph/1790361731785035776)	true	`/test pull-containerized-data-importer-e2e-ceph`
> 
> So there's definitely a bug somewhere since the image in this test should carry those environment variables.
> 
> ```
> Expected
>     <map[string]string | len:5>: {
>         "alerts.k8s.io/KubePersistentVolumeFillingUp": "disabled",
>         "app": "containerized-data-importer",
>         "app.kubernetes.io/component": "storage",
>         "app.kubernetes.io/managed-by": "cdi-controller",
>         "test.kubevirt.io/existing": "existing",
>     }
> to have {key: value}
>     <map[interface {}]interface {} | len:1>: {
>         <string>"test.kubevirt.io/test": <string>"testvalue",
> ```

https://github.com/kubevirt/containerized-data-importer/pull/3263
/test pull-containerized-data-importer-e2e-ceph

**Pull request merged**